### ISSUE TITLE
Regression after the latest multiple constructors change

### ISSUE THREAD

**Issue by mikhailshilkov** [mikhailshilkov/With.Fody#6]
@aalmada The latest constructor selection method seems to be too restrictive. I have a case when a weaver class has more properties than constructor arguments, i.e. some properties are calculated properties. Say,

``` cs
public class Sum
{
    public Sum(int a, int b)
    {
        this.A = a;
        this.B = b;
    }

    public int A { get; }
    public int B { get; }
    public int Total => this.A + this.B;
}
```

`With.Fody` fails to weave this class now, which is a regression.

**Comment by aalmada** [comment]
What are your thoughts? What about looking for the constructor with more parameters and ignore properties that are not paired? Use an attribute to ignore properties?

**Comment by mikhailshilkov** [comment]
I think we should relax the condition of all properties being present in constructor parameters, and then pick the constructor with max amount of parameters which are all properties.
I don't want to clutter code with any attributes.

**Comment by aalmada** [comment]
@mikhailshilkov Meanwhile I had started working on an attribute. I created PR #9 just so you can check how it would like. It can be ignored.

**Comment by mikhailshilkov** [comment]
Attributes, if we need them, should stay optional. I don't intend to use any attributes in our production code. I would suggest to wait with any attributes until we have a real case which can't be solved otherwise.

**Comment by aalmada** [comment]
@mikhailshilkov Implemented in #10 the fix you propose. Let me know what you think.

**Comment by aalmada** [comment]
Sorry for the regression and thanks for everything
